R/bc_bound.R

```diff
@@ -0,0 +1,166 @@
+# Copyright 2020 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+#' BC Boundary - High Resolution
+#'
+#'
+#' @param class what class you want the object in? `"sf"` (default) or `"sp"`.
+#' @param ask Should the function ask the user before downloading the data to a cache? Defaults to the value of interactive().
+#' @param force Should you force download the data?
+#'
+#' @return The spatial layer of `bc_bound_hres` in the desired class
+#'
+#'
+#' @examples
+#' \dontrun{
+#' my_layer <- bc_bound_hres()
+#' my_layer_sp <- bc_bound_hres(class = 'sp')
+#' }
+#'
+#' @source `bcdc_get_data(record = '30aeb5c1-4285-46c8-b60b-15b1a6f4258b',
+#'          resource = '3d72cf36-ab53-4a2a-9988-a883d7488384',
+#'          layer = 'BC_Boundary_Terrestrial_Multipart')`
+#'
+#' @export
+bc_bound_hres <- function(class = 'sf', ask = interactive(), force = FALSE) {
+
+  dir <- data_dir()
+  fpath <- file.path(dir, "bc_bound_hres.rds")
+
+  if (!file.exists(fpath) | force) {
+    check_write_to_data_dir(dir, ask)
+    ret <- bcdata::bcdc_get_data(record = '30aeb5c1-4285-46c8-b60b-15b1a6f4258b',
+                                 resource = '3d72cf36-ab53-4a2a-9988-a883d7488384',
+                                 layer = 'BC_Boundary_Terrestrial_Multipart')
+    ret <- rename_sf_col_to_geometry(ret)
```

Comment:
We fixed this for wfs queries. I think this is because this is gpkg. I suppose we could just create a "new" `read_sf` function (`read_sf_rename_geom`) and include that in `bcdata:::bcdc_read_functions`.

Reply:
oh right... let's not worry about that now... I think it's safe with the wfs queries, but there could be gotchas with files that we may not anticipate (what if someone puts an attribute field called `geometry` ;))